// Copyright 2020 The XLS Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package xls.fuzzer;

message NodeProto {
  // XLS op. Example: "smul".
  optional string op = 1;

  // XLS type. Possible values: "bits", "array", or "tuple".
  optional string type = 2;

  // The flattened bit width of the operation.
  optional int64 width = 3;

  // Information about the operands of the operation. Only one level of
  // recursion is supported.
  repeated NodeProto operands = 4;
}

message SampleSummaryProto {
  // Identifier for provenance of this sample. Example: "before-opt".
  optional string tag = 1;

  // XLS nodes in this IR sample.
  repeated NodeProto nodes = 3;
}

message SampleSummaries {
  repeated SampleSummaryProto samples = 1;
}
